connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/LoggingResource.java

```diff
@@ -133,20 +134,18 @@ public Response setLevel(final @PathParam("logger") String namedLogger,
         }
 
         List<Logger> childLoggers;
-        if (ROOT_LOGGER_NAME.equalsIgnoreCase(namedLogger)) {
-            childLoggers = Collections.list(currentLoggers());
+        if (ROOT_LOGGER_NAME.equalsIgnoreCase(loggerName)) {
+            childLoggers = new ArrayList<>(currentLoggers());
             childLoggers.add(rootLogger());
         } else {
             childLoggers = new ArrayList<>();
-            Logger ancestorLogger = lookupLogger(namedLogger);
-            Enumeration en = currentLoggers();
+            Logger ancestorLogger = lookupLogger(loggerName);
             boolean present = false;
-            while (en.hasMoreElements()) {
-                Logger current = (Logger) en.nextElement();
-                if (current.getName().startsWith(namedLogger)) {
-                    childLoggers.add(current);
+            for (Logger logger : currentLoggers()) {
+                if (logger.getName().startsWith(loggerName)) {
```

Comment:
It's a pre-existing issue, but I think this is slightly incorrect since it would tread `com.foo` as an ancestor of `com.foobar`. Really we should be using `startsWith` with a logger name that we know ends with a `.`.

Reply:
Agree. I am now thinking about a utility class that determines whether a given logger name is a child, descendent, parent, or ancestor of another logger.